bin/run-node

```diff
@@ -164,6 +166,10 @@ if [[ ${CARDANO_BLOCK_PRODUCER} == true ]]; then
 		--shelley-vrf-key ${CARDANO_SHELLEY_VRF_KEY} \
 		--socket-path ${CARDANO_SOCKET_PATH} \
 		--topology ${CARDANO_TOPOLOGY})
+	# Add --start-as-non-producing-node flag if START_AS_NON_PRODUCING is true
+	if [[ ${START_AS_NON_PRODUCING} == true ]]; then
+		effopts+=(--start-as-non-producing-node)
```

Comment:
Check for `--start-as-non-producing-node` and set START_AS_NON_PRODUCING in the block around line 26 to catch it as an argument being passed into the container.